core/src/test/scala/unit/kafka/server/LogRecoveryTest.scala

```diff
@@ -224,6 +224,9 @@ class LogRecoveryTest extends QuorumTestHarness {
     server1.startup()
     updateProducer()
 
+    waitUntilTrue(() => server2.replicaManager.onlinePartition(topicPartition).get.inSyncReplicaIds.size == 2,
```

Comment:
Hmm, why is this needed? Later on, there is another waitUntilTrue(). That should be enough, right?

```
    TestUtils.waitUntilTrue(() =>
      server1.replicaManager.localLogOrException(topicPartition).highWatermark == hw,
      "Failed to update high watermark for follower after timeout")
```

Also, from the jira, the failure happened in the following line.

`    leader = awaitLeaderChange(servers, topicPartition, oldLeaderOpt = Some(leader), timeout = 30000L)
`

Reply:
@junrao Can you please take a look again?